app/src/test/contrib/components/ContribNoRouteFoundPageTests.tsx

```diff
@@ -0,0 +1,36 @@
+import * as React from "react";
+import { shallow } from "enzyme";
+import { expect } from "chai";
+
+import "../../helper";
+import { Sandbox } from "../../Sandbox";
+import {createMockStore} from "../../mocks/mockStore";
+import {PageArticle} from "../../../main/shared/components/PageWithHeader/PageArticle";
+import {
+    ContribNoRouteFoundPage
+} from "../../../main/contrib/components/ContribNoRouteFoundPage";
+import {breadcrumbsActionCreators} from "../../../main/shared/actions/breadcrumbsActionsCreators";
+import {NoRouteFound} from "../../../main/shared/components/NoRouteFound";
+
+describe("Contrib No Route Found Page Component tests", () => {
+
+    const sandbox = new Sandbox();
+    afterEach(() => sandbox.restore());
+
+    it("renders component on compose level", () => {
+        let store = createMockStore();
+        const rendered = shallow(<ContribNoRouteFoundPage/>, {context: {store}});
+        expect(typeof rendered.props().createBreadcrumbs).is.equal('function');
+    });
+
+    it("renders component on component level", () => {
+        let store = createMockStore();
+        const createBreadcrumbsStub = sandbox.setStubReduxAction(breadcrumbsActionCreators, "createBreadcrumbs");
+        const rendered = shallow(<ContribNoRouteFoundPage/>, {context: {store}}).dive().dive();
+        const pageArticle = rendered.find('PageArticle');
+        expect(createBreadcrumbsStub.called).is.equal(true);
+        expect(pageArticle.props().title).is.equal(NoRouteFound.title());
+        console.log(pageArticle.debug());
```

Comment:
Huh, this `.debug()` function is useful - I didn't know it existed! But can probably remove this console log now :)

Reply:
oh, yeah, will removed that console log
i noticed that component NoRouteFound is strangely rendered
need to redo this too, but it used in too many places
so next time